enhancements/ingress/aws-lb-eip-selection.md

```diff
@@ -0,0 +1,348 @@
+// +kubebuilder:validation:XValidation:rule=`self.startsWith('eipalloc-')`,message="eipAllocations should start with 'eipalloc-'"
+// +kubebuilder:validation:XValidation:rule=`self.split("-", 2)[1].matches('[0-9a-fA-F]{17}$')`,message="eipAllocations must be 'eipalloc-' followed by exactly 17 hexadecimal characters (0-9, a-f, A-F)"
+type EIPAllocation string
+
+```
+
+### Workflow Description
+
+Pre-requisites: One must allocate an Elastic IP address from [Amazon's pool of public IPv4 addresses](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html#using-instance-addressing-eips-allocating),
+or from a custom IP address pool that you have brought to your AWS account. For more information about bringing your own IP address range to your
+AWS account, see [Bring your own IP addresses (BYOIP) in Amazon EC2](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-byoip.html)
+which could be then used in the `ingressNetworkLoadBalancer` field of the install-config.
+Note: The AWS account has some cap on the number of EIPs per region. By default, all AWS accounts have a quota of five (5) Elastic IP addresses per Region, because public (IPv4)
+internet addresses are a scarce public resource. For more details please check [Elastic IP address quota](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html#using-instance-addressing-limit).
+When this cap is hit you will get the following error: `Elastic IP address could not be allocated.
+The maximum number of addresses has been reached.`
+Elastic IP Allocation can only be done for internet-facing NLB.
+If you are using an already available eipAllocation then it should be an `UnAssociated` one.
+
+**cluster administrator** is a human user responsible for deploying a
+cluster. A cluster administrator also has the rights to modify the cluster level components.
+
+For installer-created-VPCs, a cluster admin can accurately predict how many EIPAllocations to bring so their cluster doesn't fail to install as follows as they might not 
+know what public subnets will be created by the openshift-installer.
+
+To illustrate further, the workflow could start like:
+
+1. Cluster admin wants to use EIPs, but isn't bringing their own VPC, so they need to know exactly how many EIPs to create.
+2. Cluster admin should review https://aws.amazon.com/about-aws/global-infrastructure/regions_az/ to find out how many AZs are in the region they are installing into.
+3. Cluster admin should use the same number of EIPs as AZs in their region.
+
+#### Configuring EIP for the default IngressController at installation time:
+1. Cluster administrator creates `install-config.yaml` and adds their domain, lbType to NLB, AWS region, and EIP allocation ids.
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  aws:
+    region: <AWS region>
+    lbType: NLB
+    ingressNetworkLoadBalancer:
+      eipAllocations:
```

Comment:
We discussed this change somewhere, and looks like the installer proposal[1] follows that logic: 

Considering BYO EIP can be expanded to installer re-use EIPs when installing a cluster, the install-config API change in this PR suggests to add the field `platfor.aws.eipAllocations`, introducing now the list `ingressNetworkLoadBalancer` to cover the Ingress Controller needs, but allowing expansion:
```diff
-    ingressNetworkLoadBalancer:
-      eipAllocations:
+    eipAllocations:
+      ingressNetworkLoadBalancer
```

should we need to update the EP?

cc @patrickdillon 

[1] https://github.com/openshift/installer/pull/8204/files#diff-ee27061f49f3641fa1c082879a6f0005c8986ece0519c3e286d91ae5a32947d8R2606-R2644

Reply:
It seems like that would be more accurate, but no one is working on this anymore. I will defer to @candita about how following up on this work should be handled.